src/Handler/Home.hs

```diff
@@ -1,21 +1,27 @@
 module Handler.Home where
 
-import Import
 import Yesod.Paginator
-import Yesod.Paginator.Simple
+    ( PageWidget
+    , PageWidgetConfig(..)
+    , defaultPageWidgetConfig
+    , simplePaginationWidget
+    )
+
+import Import
+import Query
 
 import Handler.ScreamDetail (singleScream)
 
 getHomeR :: Handler Html
 getHomeR = do
-    (screams, pagination) <- fetchScreams
+    (screams', pagination) <- runDB $ paginatedScreams 30 simpleWidget
+    images <- runDB $ fetchImagesForScreams screams'
+    let screams = joinOneToMany screamImage screams' images
```

Comment:
this is so close to being something applicative-able; however, seems like the lines are too dependent on other lines to accomplish it?

What if you separated this so the join function took one scream and all the images.
`joinOneToMany :: (a -> Key b) -> [Entity a] -> Entity b` then you can use `map` like so:
`map (joinOneToMany screamImage images) screams'`